packages/api/src/brevo-contact/dto/brevo-test-contact-input.factory.ts

```diff
@@ -0,0 +1,61 @@
+import { IsUndefinable } from "@comet/cms-api";
+import { Type } from "@nestjs/common";
+import { Field, InputType } from "@nestjs/graphql";
+import { Type as TypeTransformer } from "class-transformer";
+import { IsBoolean, IsNotEmpty, IsOptional, IsString, ValidateNested } from "class-validator";
+
+import { BrevoContactAttributesInterface, EmailCampaignScopeInterface } from "../../types";
+
+export interface BrevoTestContactInputInterface {
+    email: string;
+    blocked?: boolean;
+    attributes?: BrevoContactAttributesInterface;
+}
+
+export interface BrevoTestContactUpdateInputInterface {
+    blocked?: boolean;
+    attributes?: BrevoContactAttributesInterface;
+}
+
+export class BrevoTestContactInputFactory {
+    static create({
+        BrevoContactAttributes,
+        Scope,
+    }: {
+        BrevoContactAttributes?: Type<BrevoContactAttributesInterface>;
+        Scope: Type<EmailCampaignScopeInterface>;
+    }): [Type<BrevoTestContactInputInterface>] {
+        @InputType({
+            isAbstract: true,
+        })
+        class BrevoTestContactInputBase implements BrevoTestContactInputInterface {
+            @IsNotEmpty()
+            @IsString()
+            @Field()
+            email: string;
+
+            @IsBoolean()
+            @Field()
+            @IsOptional()
```

Comment:
if possible, please rather use `IsUndefinable` or `IsNullable`

(there might be a lot of places where this is not the case yet, but when starting the project IsUndefinable and IsNullable where not yet available)